modules/3d/test/test_rendering.cpp

```diff
@@ -0,0 +1,896 @@
+        }
+        break;
+        case ModelType::Clipping:
+        {
+            position = Vec3d(0.0, 0.0, 5.0);
+            lookat   = Vec3d(0.0, 0.0, 0.0);
+            upVector = Vec3d(0.0, 1.0, 0.0);
+
+            fovy = 45.0;
+
+            vertices =
+            {
+                { 2.0,  0.0, -2.0}, { 0.0, -6.0, -2.0}, {-2.0,  0.0, -2.0},
+                { 3.5, -1.0, -5.0}, { 2.5, -2.5, -5.0}, {-1.0,  1.0, -5.0},
+                {-6.5, -1.0, -3.0}, {-2.5, -2.0, -3.0}, { 1.0,  1.0, -5.0},
+            };
+
+            indices = { {0, 1, 2}, {3, 4, 5}, {6, 7, 8} };
+
+            Vec3f col1(217.0, 238.0, 185.0);
+            Vec3f col2(185.0, 217.0, 238.0);
+            Vec3f col3(150.0,  10.0, 238.0);
+
+            col1 *= (1.f / 255.f);
+            col2 *= (1.f / 255.f);
+            col3 *= (1.f / 255.f);
+
+            colors =
+            {
+                col1, col2, col3,
+                col2, col3, col1,
+                col3, col1, col2,
+            };
+        }
+        break;
+        case ModelType::Centered:
+        {
+            position = Vec3d(0.0, 0.0, 5.0);
+            lookat   = Vec3d(0.0, 0.0, 0.0);
+            upVector = Vec3d(0.0, 1.0, 0.0);
+
+            fovy = 45.0;
+
+            vertices =
+            {
+                { 2.0,  0.0, -2.0}, { 0.0, -2.0, -2.0}, {-2.0,  0.0, -2.0},
+                { 3.5, -1.0, -5.0}, { 2.5, -1.5, -5.0}, {-1.0,  0.5, -5.0},
+            };
+
+            indices = { {0, 1, 2}, {3, 4, 5} };
+
+            Vec3f col1(217.0, 238.0, 185.0);
+            Vec3f col2(185.0, 217.0, 238.0);
+
+            col1 *= (1.f / 255.f);
+            col2 *= (1.f / 255.f);
+
+            colors =
+            {
+                col1, col2, col1,
+                col2, col1, col2,
+            };
+        }
+        break;
+        case ModelType::Color:
+        {
+            position = Vec3d(0.0, 0.0, 5.0);
+            lookat   = Vec3d(0.0, 0.0, 0.0);
+            upVector = Vec3d(0.0, 1.0, 0.0);
+
+            fovy = 60.0;
+
+            vertices =
+            {
+                { 2.0,  0.0, -2.0},
+                { 0.0,  2.0, -3.0},
+                {-2.0,  0.0, -2.0},
+                { 0.0, -2.0,  1.0},
+            };
+
+            indices = { {0, 1, 2}, {0, 2, 3} };
+
+            colors =
+            {
+                { 0.0f, 0.0f, 1.0f},
+                { 0.0f, 1.0f, 0.0f},
+                { 1.0f, 0.0f, 0.0f},
+                { 0.0f, 1.0f, 0.0f},
+            };
+        }
+        break;
+
+        default:
+            break;
+        }
+    }
+
+    Vec3d position;
+    Vec3d lookat;
+    Vec3d upVector;
+
+    double fovy;
+
+    std::vector<Vec3f> vertices;
+    std::vector<Vec3i> indices;
+    std::vector<Vec3f> colors;
+};
+
+
+void compareDepth(const cv::Mat& gt, const cv::Mat& mat, cv::Mat& diff, double zFar, double scale,
+                  double maskThreshold, double normInfThreshold, double normL2Threshold)
+{
+    ASSERT_EQ(gt.type(), CV_16UC1);
```

Comment:
Signature for `_EQ` is `ASSERT_EQ(expected_reference, actual_result);`.
Correct order of parameters are required to emit valid error messages.

Reference: https://github.com/opencv/opencv/blob/4.0.0/modules/ts/include/opencv2/ts/ts_gtest.h#L8196-L8200

```
GTEST_API_ AssertionResult EqFailure(const char* expected_expression,
                                     const char* actual_expression,
                                     const std::string& expected_value,
                                     const std::string& actual_value,
                                     bool ignoring_case);
```